frontend/alert/components/common/DropdownTool.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState } from "react";
+import styled from "styled-components";
+import { Img } from "./common";
+
+interface Props {
+    show?: boolean;
+    width?: string;
+}
+
+const DropdownContainer = styled.div<Props>`
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    width: ${(props) => props.width}rem;
+    margin-right: 0.5rem;
+`;
+
+const Icon = styled(Img)`
+    padding-top: 0.05rem;
+    margin-right: 0.3rem;
+`;
+
+const Arrow = styled(Img)`
+    padding-top: 0.1rem;
+    margin-left: 0.25rem;
+`;
+
+const DropdownMenu = styled.div<Props>`
+    position: absolute;
+    display: ${({ show }) => (show ? "flex" : "none")};
+    visibility: ${({ show }) => (show ? "visible" : "hidden")};
+    flex-direction: column;
+    justify-content: center;
+    text-align: center;
+    font-size: 0.7rem;
+    font-weight: bold;
+    cursor: pointer;
+    color: #489be8;
+    overflow: hidden;
+    width: ${(props) => props.width}rem;
+    background-color: white;
+    border-radius: 0.25rem;
+    border: 0.1rem solid #e1e3e7;
+`;
+
+const DropdownTitle = styled.span`
+    display: flex;
+    font-size: 0.7rem;
+    font-weight: bold;
+    cursor: pointer;
+    color: #489be8;
+    justify-content: center;
+    text-align: center;
+    height: 1.5rem;
+    align-items: center;
+`;
+
+const DropdownItem = styled.div`
+    padding: 0.25rem 0rem 0.25rem 0rem;
+    &:hover {
+        background-color: #e1e3e7;
+    }
+`;
+
+interface DropdownToolProps {
+    actionsText: String[];
+    functions: (() => void)[];
+    width: string;
+    img: string;
+}
+
+const DropdownTool = ({
+    actionsText,
+    functions,
+    width,
```

Comment:
Why do we need to pass in `width`? Can we change `DropdownMenu`'s width to auto and add some horizontal padding to `DropdownItem`

```
const DropdownItem = styled.div`
    padding: 0.25rem 0.5rem;
    &:hover {
        background-color: #e1e3e7;
    }
`;
```

then, we can change the styling of the parent div of `DropdownMenu` to center the dropdown:
```
<div
      style={{
          position: "relative",
          display: "flex",
          justifyContent: "center",
      }}
>
```
Maybe we can also turn this div into a styled component in this case.